lib/lambda_ethereum_consensus/execution/execution_client.ex

```diff
@@ -0,0 +1,109 @@
+defmodule LambdaEthereumConsensus.Execution.ExecutionClient do
```

Comment:
maybe a Genserver is not needed here, if we decide to store all state in the config.

Reply:
I suspect we will have to store some state in the future that is not config, but we can refactor when the time comes

Reply:
This GenServer will become a bottleneck eventually. We are blocking it on RPC calls in each method, so each call will take quite a while.